tool/yubico-piv-tool.c

```diff
@@ -724,9 +724,7 @@ static bool request_certificate(ykpiv_state *state, enum enum_key_format key_for
   }
 
   memcpy(digest, oid, oid_len);
-  /* XXX: this should probably use X509_REQ_digest() but that's buggy */
-  if(!ASN1_item_digest(ASN1_ITEM_rptr(X509_REQ_INFO), md, req->req_info,
-              digest + oid_len, &digest_len)) {
+  if (!X509_REQ_digest(req, md, digest + oid_len, &digest_len)) {
```

Comment:
I didn't test this part, but I am not sure if there is an interface in the new OpenSSL to get the `req_info` from the structure. I will remove this chunk for now.
I planned to do all the changes, but some of the certificate handling was more complex than I was familiar with. Currently dumping what is available for others to have possibility to continue or I will finish that later.